src/JsonSchema/Pointer.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace JsonSchema;
+
+use JsonSchema\Exception\InvalidPointerException;
+use JsonSchema\Exception\ResourceNotFoundException;
+
+/**
+ * Resolve JSON Pointers (RFC 6901)
+ */
+class Pointer
```

Comment:
Cool :) I think it's a bit weird that this code lives in a class named `Pointer`. Objects of this class do not actually represent a pointer (e.g. `/data/2`), but the JSON data. Also, I don't understand why this class stores any state (`$json`).

In my opinion, this should be a stateless service class with a method `resolvePointer($json, $pointer)`. Following this thought, I think that this code could be just as well merged back into `UriRetriever`.